dataset/instrution_commit_clean.py

```diff
@@ -0,0 +1,325 @@
+                        f"  LONG_SAMPLING: {LONG_SAMPLING}\n" \
+                        f"  LONG_SAMPLING_THRESHOLD: {LONG_SAMPLING_THRESHOLD}\n" \
+                        f"  MD_SAMPLING: {MD_SAMPLING}\n" \
+                        f"The sampling parameters to balance the programming language as:\n" \
+                        f"  DATA_SAMPLING: {DATA_SAMPLING}\n" \
+                        f"The sampling parameters to strictly filter the dataset using it's proba of being a good instruction as:\n" \
+                        f"  PROBA_STRICT_FILTERING: {PROBA_STRICT_FILTERING}\n" \
+                        f"  STRICT_PROBA_THRESHOLD: {STRICT_PROBA_THRESHOLD}\n" \
+                        f"The sampling parameters to resample the dataset using it's proba of being a good instruction as:\n" \
+                        f"  PROBA_SOFT_RESAMPLING: {PROBA_SOFT_RESAMPLING}\n" \
+                        f"  SOFT_PROBA_THRESHOLD: {SOFT_PROBA_THRESHOLD}\n" \
+                        f"  SAMPLE_REPEAT_TIMES: {SAMPLE_REPEAT_TIMES}\n" \
+                        f"  LOW_QUALITY_SAMPLING_PROB: {LOW_QUALITY_SAMPLING_PROB}\n" \
+                        f"The sampling parameters to control the code context range as:\n" \
+                        f"  FULL_RANGE_FRAC: {FULL_RANGE_FRAC}\n" \
+                        f"  MIN_RANGE: {MIN_RANGE}\n" \
+                        f"  MAX_RANGE: {MAX_RANGE}\n"
+
+DATA_EXT = {"json", "yml", "xml", "html"}
+
+BAD_SUB_MESSAGE = [
+    "auto commit",
+    "update contributing",
+    "<?xml",
+    "merge branch",
+    "merge pull request"
+]
+
+BAD_MESSAGE = [
+    "readme",
+    "update",
+    "dummy",
+    "updated",
+    # "debug",
+    "test",
+    "update readme",
+    "update readme.md",
+    "updated readme.md",
+    "updated readme",
+
+]
+
+
+def get_line_diff_range(example):
+    old_file_start = 0
+    old_file_end = 0
+
+    new_file_start = 0
+    new_file_end = 0
+
+    n_inserts = 0
+    n_deletes = 0
+
+    for group in SequenceMatcher(None, example["old_contents"].splitlines(),
+                                 example["new_contents"].splitlines()).get_grouped_opcodes():
+        group = [g for g in group if g[0] != "equal"]
+
+        for element in group:
+            if element[0] == "insert":
+                n_inserts += element[4] - element[3]
+            if element[0] == "delete":
+                n_deletes += element[2] - element[1]
+            if element[0] == "replace":
+                n_deletes += element[2] - element[1]
+                n_inserts += element[4] - element[3]
+
+        first, last = group[0], group[-1]
+        file1_range = (first[1], last[2])
+        file2_range = (first[3], last[4])
+
+        old_file_start = min(file1_range[0], old_file_start)
+        old_file_end = max(file1_range[1], old_file_end)
+
+        new_file_start = min(file2_range[0], new_file_start)
+        new_file_end = max(file2_range[1], new_file_end)
```

Comment:
I think there's a bug here where old_file_start & new_file_start will **always** be 0 rather than the actual start of the new file, since they are initialized to 0 and the min of (x, 0) where x>=0 is always 0.

Rather, I think they should not be initialized to 0 and instead just set to the first `file1_range[0]` that occurs, no?


Indeed looking through [some examples](https://huggingface.co/datasets/bigcode/commits-pjj), I cannot find any that do not start with imports / licenses, i.e. the top of a file